diff --git a/node_modules/starlight-blog/components/Metadata.astro b/node_modules/starlight-blog/components/Metadata.astro
index d0e1da7..5334e31 100644
--- a/node_modules/starlight-blog/components/Metadata.astro
+++ b/node_modules/starlight-blog/components/Metadata.astro
@@ -22,7 +22,7 @@ const hasAuthors = authors.length > 0
 
 let metrics: Metrics | undefined
 
-if (config.metrics.readingTime !== false || config.metrics.words !== false) {
+if (config !== undefined && config.metrics != undefined && (config.metrics.readingTime !== false || config.metrics.words !== false)) {
   const html = await renderBlogEntryToString(entry, Astro.locals.t)
   metrics = getMetrics(html, Astro.currentLocale, entry.data.metrics)
 }
@@ -47,14 +47,14 @@ if (config.metrics.readingTime !== false || config.metrics.words !== false) {
       ) : null
     }
     {
-      config.metrics.readingTime !== false && metrics && metrics.readingTime.minutes > 0 && (
+      config !== undefined && config.metrics !== undefined && config.metrics.readingTime !== false && metrics && metrics.readingTime.minutes > 0 && (
         <span class="metadata-alt">
           {Astro.locals.t('starlightBlog.metrics.readingTime.minutes', { count: metrics.readingTime.minutes })}
         </span>
       )
     }
     {
-      config.metrics.words !== false && metrics && metrics.words.total > 0 && (
+      config !== undefined && config.metrics !== undefined && config.metrics.words !== false && metrics && metrics.words.total > 0 && (
         <span class="metadata-alt">
           {Astro.locals.t('starlightBlog.metrics.words', {
             count: config.metrics.words === 'total' ? metrics.words.total : metrics.words.rounded,
diff --git a/node_modules/starlight-blog/components/PostTags.astro b/node_modules/starlight-blog/components/PostTags.astro
index 4b31304..cbf7444 100644
--- a/node_modules/starlight-blog/components/PostTags.astro
+++ b/node_modules/starlight-blog/components/PostTags.astro
@@ -2,7 +2,7 @@
 import type { StarlightBlogConfig } from '../libs/config'
 import type { StarlightBlogEntry } from '../libs/content'
 import type { Locale } from '../libs/i18n'
-import { getRelativeBlogUrl } from '../libs/page'
+import { getRelativeUrl } from '../libs/page'
 import { getEntryTags } from '../libs/tags'
 
 interface Props {
@@ -23,7 +23,7 @@ const tags = getEntryTags(entry)
       <ul>
         {tags.map((tag) => (
           <li>
-            <a href={getRelativeBlogUrl(config, `/tags/${tag.slug}`, locale)}>{tag.label}</a>
+            <a href={getRelativeUrl(entry.id, `/tags/${tag.slug}`, locale)}>{tag.label}</a>
           </li>
         ))}
       </ul>
diff --git a/node_modules/starlight-blog/components/Preview.astro b/node_modules/starlight-blog/components/Preview.astro
index 778cc68..0434584 100644
--- a/node_modules/starlight-blog/components/Preview.astro
+++ b/node_modules/starlight-blog/components/Preview.astro
@@ -1,6 +1,6 @@
 ---
 import type { StarlightBlogPostData } from '../data'
-import { getBlogConfigFromPath, getLocaleFromRelativeUrl } from '../libs/page'
+import { getBlogConfigFromPath, getLocaleFromRelativeUrl, getRelativeUrl, getPathWithLocale } from '../libs/page'
 
 import Cover from './Cover.astro'
 import Excerpt from './Excerpt.astro'
@@ -21,25 +21,45 @@ if (!config) throw new Error(`Failed to get blog configuration for entry '${entr
 ---
 
 <article class="preview">
-  <header>
-    <a class="preview-link" href={post.href}>
-      <h2>{entry.data.title}</h2>
+  <div class="preview-left">
+    <header>
+      <h2>
+        <a href={getRelativeUrl(`/${getPathWithLocale(entry.id, locale)}`)}>{entry.data.title}</a>
+      </h2>
+      <Metadata entry={entry} {locale} />
+    </header>
+    <div class="sl-markdown-content">
+      <Excerpt {entry} />
+    </div>
+    <footer>
+      <PostTags {entry} {locale} />
+    </footer>
+  </div>
+    <div class="preview-right">
       {entry.data.cover && <Cover cover={entry.data.cover} alt="" />}
-    </a>
-    <Metadata {config} {entry} />
-  </header>
-  <div class="sl-markdown-content">
-    <Excerpt {entry} />
   </div>
-  <footer>
-    <PostTags {config} {entry} {locale} />
-  </footer>
 </article>
 
 <style>
   .preview {
+    display: flex;
+    gap: 1.5rem;
+  }
+
+  .preview-left {
+    display: flex;
+    flex-direction: column;
+    flex-grow: 1;
+    width: 50%;
+    gap: 1.5rem;
+  }
+
+  .preview-right {
     display: flex;
     flex-direction: column;
+    width: 50%;
+    height: 100%;
+    object-fit: cover;
     gap: 1.5rem;
   }
 
diff --git a/node_modules/starlight-blog/components/TagList.astro b/node_modules/starlight-blog/components/TagList.astro
new file mode 100644
index 0000000..1838a9a
--- /dev/null
+++ b/node_modules/starlight-blog/components/TagList.astro
@@ -0,0 +1,33 @@
+import LinkCard from '@astrojs/starlight/components'
+import type { Locale } from '../libs/i18n'
+import {getAllTags} from '../libs/tags'
+import {getPathWithLocale, getRelativeUrl, isBlogTagPage} from '../libs/page'
+
+interface Props {
+        entries: StarlightBlogEntry[]
+        locale: Locale
+    }
+
+const { entries, locale } = Astro.props
+
+const tags = await getAllTags(locale)
+---
+
+<div class="posts">
+    { tags.prototype.forEach((tagKey, tagValue, tagMap) => (
+        <LinkCard title={tagKey} href={
+            getRelativeUrl(
+                getPathWithLocale(`/tags/${tagKey}`, locale),
+                false
+            )} />
+        ))
+    }
+</div>
+
+<style>
+    .posts {
+        display: flex;
+        flex-direction: column;
+        gap: 3rem;
+    }
+</style>
\ No newline at end of file
diff --git a/node_modules/starlight-blog/index.ts b/node_modules/starlight-blog/index.ts
index 23d2d09..040a398 100644
--- a/node_modules/starlight-blog/index.ts
+++ b/node_modules/starlight-blog/index.ts
@@ -93,19 +93,7 @@ export default function starlightBlogPlugin(userConfig?: StarlightBlogUserConfig
             'astro:config:setup': ({ injectRoute, updateConfig }) => {
               injectRoute({
                 entrypoint: 'starlight-blog/routes/Tags.astro',
-                pattern: '/[...prefix]/tags/[tag]',
-                prerender: true,
-              })
-
-              injectRoute({
-                entrypoint: 'starlight-blog/routes/Authors.astro',
-                pattern: '/[...prefix]/authors/[author]',
-                prerender: true,
-              })
-
-              injectRoute({
-                entrypoint: 'starlight-blog/routes/Blog.astro',
-                pattern: '/[...prefix]/[...page]',
+                pattern: '/tags/[tag]',
                 prerender: true,
               })
 
diff --git a/node_modules/starlight-blog/libs/authors.ts b/node_modules/starlight-blog/libs/authors.ts
index 87266b8..5c40042 100644
--- a/node_modules/starlight-blog/libs/authors.ts
+++ b/node_modules/starlight-blog/libs/authors.ts
@@ -48,7 +48,7 @@ export async function getAuthorsStaticPaths() {
 export function getEntryAuthors(config: StarlightBlogConfig, entry: StarlightBlogEntry): StarlightBlogAuthor[] {
   const authors: StarlightBlogAuthor[] = []
 
-  if (!entry.data.authors) {
+  /*if (!entry.data.authors) {
     authors.push(...Object.values(config.authors))
   } else if (typeof entry.data.authors === 'string') {
     authors.push(getAuthorFromConfig(config, entry.data.authors))
@@ -62,7 +62,7 @@ export function getEntryAuthors(config: StarlightBlogConfig, entry: StarlightBlo
     }
   } else {
     authors.push(entry.data.authors)
-  }
+  }*/
 
   return authors
 }
diff --git a/node_modules/starlight-blog/libs/config.ts b/node_modules/starlight-blog/libs/config.ts
index c128492..09177ad 100644
--- a/node_modules/starlight-blog/libs/config.ts
+++ b/node_modules/starlight-blog/libs/config.ts
@@ -47,7 +47,7 @@ const blogConfigSchema = z
      *
      * @default 'header-end'
      */
-    navigation: z.union([z.literal('header-start'), z.literal('header-end'), z.literal('none')]).default('header-end'),
+    navigation: z.union([z.literal('header-start'), z.literal('header-end'), z.literal('sidebar'), z.literal('none')]).default('header-end'),
     /**
      * The base prefix for all blog routes.
      *
diff --git a/node_modules/starlight-blog/libs/navigation.ts b/node_modules/starlight-blog/libs/navigation.ts
index 2fbbecc..02b9f26 100644
--- a/node_modules/starlight-blog/libs/navigation.ts
+++ b/node_modules/starlight-blog/libs/navigation.ts
@@ -3,7 +3,7 @@ import type { StarlightUserConfig } from '@astrojs/starlight/types'
 import type { StarlightBlogConfig } from './config'
 
 export function isNavigationWithSidebarLink(config: StarlightBlogConfig) {
-  return config.navigation === 'header-start' || config.navigation === 'header-end'
+  return config.navigation === 'sidebar'
 }
 
 export function isNavigationWithCustomCss(config: StarlightBlogConfig) {
diff --git a/node_modules/starlight-blog/libs/page.ts b/node_modules/starlight-blog/libs/page.ts
index 098cd52..d339327 100644
--- a/node_modules/starlight-blog/libs/page.ts
+++ b/node_modules/starlight-blog/libs/page.ts
@@ -54,7 +54,7 @@ export function getRelativeBlogUrl(
 }
 
 /** Get the relative URL of a page taking into account the base URL and the trailing slash configuration. */
-export function getRelativeUrl(path: string, ignoreTrailingSlash = false) {
+export function getRelativeUrl(path, ignoreTrailingSlash = false) {
   path = stripLeadingSlash(path)
   path = path ? `${base}/${path}` : `${base}/`
 
diff --git a/node_modules/starlight-blog/libs/path.ts b/node_modules/starlight-blog/libs/path.ts
index 9bbe86c..eb440ff 100644
--- a/node_modules/starlight-blog/libs/path.ts
+++ b/node_modules/starlight-blog/libs/path.ts
@@ -1,4 +1,6 @@
-export function stripLeadingSlash(path: string) {
+export function stripLeadingSlash(path) {
+  console.log(path)
+  console.log(typeof path)
   if (!path.startsWith('/')) {
     return path
   }
diff --git a/node_modules/starlight-blog/libs/rss.ts b/node_modules/starlight-blog/libs/rss.ts
index e054963..18a232d 100644
--- a/node_modules/starlight-blog/libs/rss.ts
+++ b/node_modules/starlight-blog/libs/rss.ts
@@ -73,7 +73,9 @@ export async function getRSSOptions(
   )
 }
 
-export async function getRSSArchiveOptions(
+export async function 
+
+getRSSArchiveOptions(
   config: StarlightBlogConfig,
   site: URL | undefined,
   locale: Locale,
diff --git a/node_modules/starlight-blog/libs/tags.ts b/node_modules/starlight-blog/libs/tags.ts
index f7c958a..5689062 100644
--- a/node_modules/starlight-blog/libs/tags.ts
+++ b/node_modules/starlight-blog/libs/tags.ts
@@ -58,7 +58,7 @@ function getTagsStaticPath(
 ) {
   return {
     params: {
-      prefix: getPathWithLocale(config.prefix, locale),
+      prefix: getPathWithLocale('', locale),
       tag: slug,
     },
     props: {
diff --git a/node_modules/starlight-blog/middleware.ts b/node_modules/starlight-blog/middleware.ts
index 2ccc911..28c703f 100644
--- a/node_modules/starlight-blog/middleware.ts
+++ b/node_modules/starlight-blog/middleware.ts
@@ -125,7 +125,8 @@ async function getBlogPostsData(
       const tags = getEntryTags(entry)
       const html = await renderBlogEntryToString(entry, t)
       const metrics = getMetrics(html, locale, entry.data.metrics)
-
+      console.log("GET BLOG POSTS DATA")
+      console.log(config)
       const postsData: StarlightBlogData['posts'][number] = {
         authors: authors.map(({ name, title, url }) => ({
           name,
@@ -141,7 +142,7 @@ async function getBlogPostsData(
         metrics,
         tags: tags.map(({ label, slug }) => ({
           label,
-          href: getRelativeBlogUrl(config, `/tags/${slug}`, locale),
+          href: getRelativeUrl(`/${getPathWithLocale(entry.id, locale)}`, `/tags/${slug}`, locale),
         })),
         title: entry.data.title,
       }
diff --git a/node_modules/starlight-blog/translations.ts b/node_modules/starlight-blog/translations.ts
index 1804ef8..7b198e5 100644
--- a/node_modules/starlight-blog/translations.ts
+++ b/node_modules/starlight-blog/translations.ts
@@ -12,12 +12,12 @@ export const Translations = {
       ' - Last update: <time datetime="{{isoDate}}">{{date, datetime(dateStyle: medium)}}</time>',
     'starlightBlog.post.draft': 'Draft',
     'starlightBlog.post.featured': 'Featured',
-    'starlightBlog.post.tags': 'Tags:',
-    'starlightBlog.sidebar.all': 'All posts',
-    'starlightBlog.sidebar.featured': 'Featured posts',
-    'starlightBlog.sidebar.recent': 'Recent posts',
-    'starlightBlog.sidebar.tags': 'Tags',
-    'starlightBlog.sidebar.authors': 'Authors',
+    'starlightBlog.post.tags': 'tags:',
+    'starlightBlog.sidebar.all': 'All Posts',
+    'starlightBlog.sidebar.featured': 'featured posts',
+    'starlightBlog.sidebar.recent': 'recent posts',
+    'starlightBlog.sidebar.tags': 'tags',
+    'starlightBlog.sidebar.authors': 'authors',
     'starlightBlog.sidebar.rss': 'RSS',
     'starlightBlog.tags.count_one': '{{count}} post with the tag “{{tag}}”',
     'starlightBlog.tags.count_other': '{{count}} posts with the tag “{{tag}}”',
